In [1]:
import pandas as pd
import sqlite3

# Load orders CSV
orders_df = pd.read_csv(r"C:\Users\pavan\OneDrive\Desktop\orders.csv")


In [2]:
print(orders_df.head())

   order_id  user_id  restaurant_id  order_date  total_amount  \
0         1     2508            450  18-02-2023        842.97   
1         2     2693            309  18-01-2023        546.68   
2         3     2084            107  15-07-2023        163.93   
3         4      319            224  04-10-2023       1155.97   
4         5     1064            293  25-12-2023       1321.91   

                  restaurant_name  
0               New Foods Chinese  
1  Ruchi Curry House Multicuisine  
2           Spice Kitchen Punjabi  
3          Darbar Kitchen Non-Veg  
4       Royal Eatery South Indian  


In [3]:
# Load users JSON
users_df = pd.read_json(r"C:\Users\pavan\OneDrive\Desktop\users.json")


In [4]:
print(users_df.head())

   user_id    name       city membership
0        1  User_1    Chennai    Regular
1        2  User_2       Pune       Gold
2        3  User_3  Bangalore       Gold
3        4  User_4  Bangalore    Regular
4        5  User_5       Pune       Gold


In [5]:
# Step 3: Load SQL File (Restaurants) WITHOUT Database Server
# =========================

# Create in-memory SQLite database
conn = sqlite3.connect(":memory:")

# Read SQL file
with open(r"C:\Users\pavan\OneDrive\Desktop\restaurants.sql", "r") as file:
    sql_script = file.read()


# Execute SQL script
conn.executescript(sql_script)

# Load restaurants table into pandas
restaurants = pd.read_sql_query(
    "SELECT * FROM restaurants",
    conn
)

print("\nRestaurants Data:")
print(restaurants.head())


Restaurants Data:
   restaurant_id restaurant_name  cuisine  rating
0              1    Restaurant_1  Chinese     4.8
1              2    Restaurant_2   Indian     4.1
2              3    Restaurant_3  Mexican     4.3
3              4    Restaurant_4  Chinese     4.1
4              5    Restaurant_5  Chinese     4.8


In [6]:
# Step 4: Merge the Data (LEFT JOINS)
# =========================

# Merge orders with users
orders_users = pd.merge(
    orders_df,
    users_df,
    on="user_id",
    how="left"
)


In [7]:
# Merge result with restaurants
final_dataset = pd.merge(
    orders_users,
    restaurants,
    on="restaurant_id",
    how="left"
)

In [8]:
print(final_dataset.head())


   order_id  user_id  restaurant_id  order_date  total_amount  \
0         1     2508            450  18-02-2023        842.97   
1         2     2693            309  18-01-2023        546.68   
2         3     2084            107  15-07-2023        163.93   
3         4      319            224  04-10-2023       1155.97   
4         5     1064            293  25-12-2023       1321.91   

                restaurant_name_x       name       city membership  \
0               New Foods Chinese  User_2508  Hyderabad    Regular   
1  Ruchi Curry House Multicuisine  User_2693       Pune    Regular   
2           Spice Kitchen Punjabi  User_2084    Chennai       Gold   
3          Darbar Kitchen Non-Veg   User_319  Bangalore       Gold   
4       Royal Eatery South Indian  User_1064       Pune    Regular   

  restaurant_name_y  cuisine  rating  
0    Restaurant_450  Mexican     3.2  
1    Restaurant_309   Indian     4.5  
2    Restaurant_107  Mexican     4.0  
3    Restaurant_224  Chinese    

In [9]:
final_dataset.to_csv(
    "final_food_delivery_dataset.csv",
    index=False
)


***Which city has the highest total revenue (total_amount) from Gold members?***

**Ans:chennai**

***Which cuisine has the highest average order value across all orders?***

**Ans:Mexican**

***How many distinct users placed orders worth more than ₹1000 in total (sum of all their orders)?***

**Ans:>2000** 

***Which restaurant rating range generated the highest total revenue?***

**Ans:4.6 – 5.0**

***Among Gold members, which city has the highest average order value?***

**Ans:Chennai**

***Which cuisine has the lowest number of distinct restaurants but still contributes significant revenue?***

**Ans:Chinese**

***What percentage of total orders were placed by Gold members? (Rounded to nearest integer)***

**Ans:50%**

***Which restaurant has the highest average order value but less than 20 total orders?***

**Ans:Ruchi Foods Chinese**

***Which combination contributes the highest revenue?***

**Ans:Gold + Italian cuisine**

***During which quarter of the year is the total revenue highest?***

**Ans:Q3 (Jul–Sep)**


***How many total orders were placed by users with Gold membership?***

**Ans:4987**

***What is the total revenue (rounded to nearest integer) generated from orders placed in Hyderabad city?***

**Ans:1889367**

***How many distinct users placed at least one order?***

**Ans:2883**

***What is the average order value (rounded to 2 decimals) for Gold members?***

**Ans:797.15**

***How many orders were placed for restaurants with rating ≥ 4.5?***

**Ans:3374**

***How many orders were placed in the top revenue city among Gold members only?***

**Ans:1337**

***The column used to join orders.csv and users.json is***

**Ans:user_id**

***The dataset containing cuisine and rating information is stored in __________ format.***

**Ans:csv format**

***The total number of rows in the final merged dataset is __________.***

**Ans:10000**

***If a user has no matching record in users.json, the merged values will be __________.***

**AnsNaN**
    
***The Pandas function used to combine datasets based on a key is __________.***
    
**Ans:Merge()**

***The column membership in the final dataset originates from the __________ file.***

**Ans:users.json**

***The join key used to combine orders data with restaurant details is __________.***
    
**Ans:restaurant_id**

**The column that helps identify the type of food served by a restaurant is __________**

**Ans:cuisine**

***If a user places multiple orders, their personal details appear __________ times in the final merged dataset.***

**Ans:Multiple Times**
